swap/src/asb/event_loop.rs

```diff
@@ -479,16 +479,38 @@ where
         // use unlocked monero balance for quote
         let xmr_balance = Amount::from_piconero(balance.unlocked_balance);
 
-        let max_bitcoin_for_monero =
-            xmr_balance
-                .max_bitcoin_for_price(ask_price)
-                .ok_or_else(|| {
-                    anyhow!(
-                        "Bitcoin price ({}) x Monero ({}) overflow",
-                        ask_price,
-                        xmr_balance
-                    )
-                })?;
+        // From our full balance we need to subtract any Monero that is 'reserved' for ongoing swaps
+        // (where the Bitcoin has been (or is being) locked but we haven't sent the Monero yet).
+        //
+        // TODO: Better manage monero funds. Currently we store all Monero in one UTXO, then the change
+        // address is blocked for 10 blocks before we can use it again.
+        // We should distribute the monero funds into multiple UTXOs, to avoid blocking large amounts of the total balance.
+
+        let reserved: Amount = self
+            .db
+            .all()
+            .await?
+            .iter()
+            .filter_map(|(_, state)| match state {
+                State::Alice(AliceState::BtcLockTransactionSeen { state3 })
+                | State::Alice(AliceState::BtcLocked { state3 }) => Some(state3.xmr + MONERO_FEE),
+                _ => None,
+            })
+            .fold(Amount::ZERO, |acc, amount| acc + amount);
+
+        let free_monero_balance = xmr_balance
+            .checked_sub(reserved)
+            .context("reserved more monero than we've got")?;
```

Comment:
I think we should fallback to responding with a `max_swap_amount` of `0` here. And warn the user about the missing funds?

This might actually happen from time to time and not responding with a quote at all is not good behaviour.

Reply:
Fixed that.